### ISSUE TITLE
26 is an unsupported value for <enum 'MeasurementType

### ISSUE THREAD

**Issue by Bluhme1** [joostlek/python-withings#304]
`2024-06-28 13:38:41.403 DEBUG (MainThread) [homeassistant.components.withings] Finished fetching Withings 17043154 sleep data in 0.453 seconds (success: True)
2024-06-28 13:38:41.538 DEBUG (MainThread) [homeassistant.components.withings] Finished fetching Withings 17043154 workout data in 0.135 seconds (success: True)
2024-06-28 13:38:41.539 DEBUG (MainThread) [homeassistant.components.withings] Finished fetching Withings 17043154 bed presence data in 0.000 seconds (success: True)
2024-06-28 13:38:41.659 DEBUG (MainThread) [homeassistant.components.withings] Finished fetching Withings 17043154 activity data in 0.120 seconds (success: True)
2024-06-28 13:38:41.738 DEBUG (MainThread) [homeassistant.components.withings] Finished fetching Withings 17043154 goals data in 0.079 seconds (success: True)
2024-06-28 13:38:41.922 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.923 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.923 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.924 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.924 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.924 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.925 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.925 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.926 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.928 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.929 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.930 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.930 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.931 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.931 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.932 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.933 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.933 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.934 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.934 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.935 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.935 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.936 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.937 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.937 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.937 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.938 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.939 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.940 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.940 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.941 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.941 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.942 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.943 WARNING (MainThread) [aiowithings] 226 is an unsupported value for <enum 'MeasurementType'>, please report this at https://github.com/joostlek/python-withings/issues
2024-06-28 13:38:41.945 DEBUG (MainThread) [homeassistant.components.withings] Finished fetching Withings 17043154 measurements data in 0.207 seconds (success: True)
2024-06-28 13:38:42.949 DEBUG (MainThread) [homeassistant.components.withings] Registered Withings webhook at hass: https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0=
2024-06-28 13:38:43.088 DEBUG (MainThread) [homeassistant.components.withings] Unsubscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 1 in 1.0 seconds
2024-06-28 13:38:44.161 DEBUG (MainThread) [homeassistant.components.withings] Unsubscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 4 in 1.0 seconds
2024-06-28 13:38:45.240 DEBUG (MainThread) [homeassistant.components.withings] Unsubscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 16 in 1.0 seconds
2024-06-28 13:38:46.311 DEBUG (MainThread) [homeassistant.components.withings] Unsubscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 44 in 1.0 seconds
2024-06-28 13:38:47.392 DEBUG (MainThread) [homeassistant.components.withings] Unsubscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 50 in 1.0 seconds
2024-06-28 13:38:48.467 DEBUG (MainThread) [homeassistant.components.withings] Unsubscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 51 in 1.0 seconds
2024-06-28 13:38:49.546 DEBUG (MainThread) [homeassistant.components.withings] Subscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 1 in 5.0 seconds
2024-06-28 13:38:54.822 DEBUG (MainThread) [homeassistant.components.withings] Subscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 4 in 5.0 seconds
2024-06-28 13:39:00.144 DEBUG (MainThread) [homeassistant.components.withings] Subscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 44 in 5.0 seconds
2024-06-28 13:39:05.443 DEBUG (MainThread) [homeassistant.components.withings] Subscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 16 in 5.0 seconds
2024-06-28 13:39:10.738 DEBUG (MainThread) [homeassistant.components.withings] Subscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 50 in 5.0 seconds
2024-06-28 13:39:16.202 DEBUG (MainThread) [homeassistant.components.withings] Subscribing https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0= for 51 in 5.0 seconds
2024-06-28 13:39:21.473 DEBUG (MainThread) [homeassistant.components.withings] Registered Withings webhook at Withings: https://hooks.nabu.casa/gAAAAABlHURFN3IGRR8Y4oWT1bbWz-GzQg-9KP4eq-Biz-DClJzRpZ0QIPPFxwkzD_YKicC0A1MDUKEUDTj8zUQyAwrCg4uXjXUrPIBULNkNZ4gKJ9jXPLN_njrQ63gwvgobHGRYvAuNsfYOJfFx9RNPx82Dre7QMjTUMXj5WoWh6-BRNXy42l0=

**Comment by joostlek** [comment]
Duplicate of #157, I already reached out to Withings but no response yet. Thanks for reporting

**Comment by Bluhme1** [comment]
@joostlek 

But was working before beta. So the beta changes #119126 and #119128 or #118854 must be the reason

**Comment by joostlek** [comment]
I don't thing that that would be the problem as this issue is logged in the library, and the way we call the library didn't change. The library didn't have changes on this so I don't believe that this is caused because of the beta changes

**Comment by Bluhme1** [comment]
@joostlek I am sorry but I don't agree with you. I have reverted back to 2024.6.4 and the problem disappeared. Now I get readings from all my sensors connected to Withings.
I am sure that more people will experience the same come next Wednesday hen there is an official release and more people will upgrade.
If I can help in any way I will be glad to.
/Lars

**Comment by joostlek** [comment]
> Now I get readings from all my sensors connected to Withings.

You didn't get with the beta?

**Comment by Bluhme1** [comment]
No I didn't get readings from "Bone mass", "Fat mass" "Hydration", "Fat free mass""Muscle mass" and "Withings visceral fat index".
I have the Body comp scale

I sent you an error log from the beta-version

**Comment by joostlek** [comment]
I think there are 2 things going on
1. those lines in the log. They have been here since a long time, you can check that other issue, but I don't believe they started coming since this release. And they are not harmless, they are just there for me to collect new data to poke withings for to get new ones.
2. What is interesting is that the rest doesn't work. Can you provide diagnostics?

**Comment by joostlek** [comment]
Oh and the integrations works fine for me in the beta, so I can assure that there hasn't been any parsing regression

**Comment by Bluhme1** [comment]
Have you got the Body Comp scale.

I will go back to beta version to download diagnostics. Hang on, please

**Comment by joostlek** [comment]
I have an old one, but it at least confirms that its not a bug with all measurements.

It's strange, I got fixtures from users and I made unit tests around them, so I am pretty confident that it works like expected, but I am happy to dive into this

**Comment by Bluhme1** [comment]
Great

Here is my diagnostic file. Hope it can help you find the reason

[config_entry-withings-45cac57af5d8f088af2cb3796b2032c4.json](https://github.com/user-attachments/files/16030867/config_entry-withings-45cac57af5d8f088af2cb3796b2032c4.json)

**Comment by joostlek** [comment]
```json

    "received_measurements": {
      "positional": {
        "fat_free_mass": [
          "between_legs"
        ],
        "fat_mass_weight": [
          "between_legs"
        ],
        "muscle_mass": [
          "between_legs"
        ],
        "hydration": [
          "between_legs"
        ],
        "bone_mass": [
          "between_legs"
        ],
        "visceral_fat": [
          "between_legs"
        ],
        "unknown": [
          "between_legs"
        ]
      },
      "non_positional": [
        "pulse_wave_velocity",
        "vascular_age",
        "heart_rate",
        "nerve_health_score_feet",
        "weight",
        "fat_ratio",
        "diastolic_blood_pressure",
        "systolic_blood_pressure",
        "atrial_fibrillation"
      ]
    },
```
For some reason some non positional measurements are communicated as positional. Apparently Withings is not consistent with this because in the fixture I got from someone else it would register them on whole_body instead of between_legs.

**Comment by Bluhme1** [comment]
Ok, that might explain it. Can it be solved without ruining other readings?

**Comment by joostlek** [comment]
I have to check what the other response was. But I first need to get lunch (or breakfast), its already 4pm :x

**Comment by Bluhme1** [comment]
of course. Have a nice lunch - though it is a little late

**Comment by Bluhme1** [comment]
@joostlek 
Thank you very much. I can see that you have made a PR #305 and merged it into python-withings.

Excuse my ignorance, but how will it be reflected in Home Assistant 2024.7. The next beta release??

**Comment by joostlek** [comment]
It will be in the next beta

**Comment by Bluhme1** [comment]
Great. Thank you

**Comment by Bluhme1** [comment]
With 2024.7.b5 everything works fine again.
I own  the Body Comp scale, the BMP Core and the Sleep device.
Would be happy to help testing beta changes / improvements again.
/Lars . Denmark

**Comment by joostlek** [comment]
Lovely. I dont have any more changes planned at the moment, its just waiting on withings to get more information. 

But thanks for helping out!
